src/Symfony/Component/Serializer/Normalizer/UidNormalizer.php

```diff
@@ -87,6 +93,10 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
      */
     public function supportsDenormalization(mixed $data, string $type, string $format = null): bool
     {
+        if (AbstractUid::class === $type) {
+            trigger_deprecation('symfony/serializer', '6.1', sprintf('Supporting denormalization for the "%s" type in "%s" is deprecated, use one of "%s" child class instead.', AbstractUid::class, __CLASS__, AbstractUid::class));
```

Comment:
let's `return true` in this `if`, and use `is_subclass_of` below, to make the code ready for v7